### PR TITLE
Add registration flow

### PR DESCRIPTION
**Pull request by gsvr**

### CODE REVIEW

File: go-app.js [hunk 1]
```diff
@@ -8,14 +8,64 @@ go;
...
+    var PaginatedState = vumigo.states.PaginatedState;
     var EndState = vumigo.states.EndState;
 
+
+    go.utils = {
+
+        timed_out: function(im) {
+            return im.msg.session_event === 'new'
+                && im.user.state.name
+                && im.user.state.name !== 'state_language'
```
Comment (imsickofmaps) [inline go-app.js]:
```
no_redirects = ['state_language', 'state_migrant_main', 'state_refugee_main'];
no_redirects.indexOf(im.user.state.name) !== -1
```

would be cleaner

File: go-app.js [hunk 2]
```diff
@@ -42,26 +95,310 @@ go.app = function() {
...
-                    new Choice('state_start', $('Show this menu again')),
-                    new Choice('state_end', $('Exit'))],
+                    new Choice('english', $("English")),
+                    new Choice('french', $("French")),
+                    new Choice('amharic', $("Amharic")),
+                    new Choice('swahili', $("Swahili")),
+                    new Choice('somail', $("Somail")),
+                ],
+                next: function(choice) {
+                    self.contact.extra.language = choice.value;
```
Comment (imsickofmaps) [inline go-app.js]: add an extra for `lang` that has the ISO code, we need that for translation POT files matching. Then add a mapper and save the two extras.

**Comment by gsvr** [comment]
@imsickofmaps Please re-review.

**Comment by imsickofmaps** [comment]
:+1:

**Pull request merged**